app/code/community/Drip/Connect/Helper/Quote.php

```diff
@@ -132,8 +132,20 @@ protected function prepareQuoteItemsData(Mage_Sales_Model_Quote $quote)
             }
 
             $productVariantItem = $item;
+            $productVariantProduct = $product;
             if ($item->getProductType() === 'configurable' && $childItems[$item->getId()]) {
                 $productVariantItem = $childItems[$item->getId()];
+                $productVariantProduct = Mage::getModel('catalog/product')->load($productVariantItem->getProduct()->getId());
+            }
+
+            $productImage = $productVariantProduct->getImage();
+
+            if (empty($productImage) || $productVariantProduct->getVisibility() == Mage_Catalog_Model_Product_Visibility::VISIBILITY_NOT_VISIBLE) {
```

Comment:
@will-in-wi I got all tests passing. A little investigation revealed that `getVisibility()` is returning a string (`"1"`) while the constant is an integer, so the `===` comparison was failing. I swapped it for `==`, but please let me know if there's a better way to coerce and compare here.

Reply:
Ah… That makes sense. I don't know if it's better, but the alternate would be:
```suggestion
            if (empty($productImage) || (int)$productVariantProduct->getVisibility() === Mage_Catalog_Model_Product_Visibility::VISIBILITY_NOT_VISIBLE) {
```

But I'd probably do what you are doing.